src/main/kotlin/three/consulting/epoc/service/CustomerService.kt

```diff
@@ -16,4 +16,17 @@ class CustomerService(private val customerRepository: CustomerRepository) {
         }
         return null
     }
+
+    fun createCustomer(customerData: CustomerDTO): CustomerDTO? {
+        if (customerData.id == null) {
+            val customer = Customer(customerData)
+            val savedCustomer: Customer? = customerRepository.save(customer)
+            if (savedCustomer != null) {
+                return CustomerDTO(savedCustomer)
+            }
+        }
+        else throw UnableToCreateCustomer()
+        return null
```

Comment:
You'd never arrive in this return so it can be removed. Either return the saved customer above or throw the exception in the else when the id wasn't null to begin with